### DISCUSSION TITLE
Inconsistent Object Properties

### DISCUSSION THREAD

**Discussion by JurgenZiemer** [edmcouncil/fibo#1412] [General] [answered]
Pertains to FIBO Development: https://spec.edmcouncil.org/fibo/ontology/master/latest/dev.ttl.zip
Example Source file: \DER\ExchangeTradedDerivatives\ExchangeTradedOptions\ExchangeTradedOptions.ttl

The _localnames_ of the object properties below are already used as class names.
```
fibo-bp-iss-ipo:syndicateMember
fibo-bp-iss-muni:subscriber
fibo-der-etd-eto:strikePrice
fibo-fnd-txn-mkt:counterparty
fibo-fnd-txn-mkt:principal
fibo-fnd-txn-sup:customer
fibo-fnd-txn-sup:vendor
fibo-loan-loan-loan:loanRetainedAmount
fibo-sec-dbt-ab-syn:trancheType
```
This causes an error in my PowerDesigner data modeling tool import. The process transforms object properties into data model associative entities, which already exist. It may also impact other consumers of machine-readable format, depending on unique qnames/localnames.  

The properties don't follow the FIBO naming convention to start with a verb. (e.g. fibo-fnd-acc-cur:hasPrice)
A closer look at _strikePrice_ in ExchangeTradedOptions, for example, has several properties with a number at the end:

```
fibo-der-etd-eto:strikePrice
	a owl:ObjectProperty ; (line 752)
        rdfs:domain fibo-der-etd-eto:CurrencyOptionExerciseTerms ;

fibo-der-etd-eto:strikePrice.1
	a owl:ObjectProperty ; (line 762)
        rdfs:domain fibo-der-etd-eto:FutureOptionsExerciseTerms ;

fibo-der-etd-eto:strikePrice.2
	a owl:DatatypeProperty ; (line 772)
        rdfs:domain fibo-der-etd-eto:OptionInstrumentExerciseTerms ;

fibo-der-etd-eto:strikePrice.3
	a owl:ObjectProperty ; (line 782)
	rdfs:domain fibo-der-etd-eto:StockOptionExerciseTerms ;
```
The datatype property, `strikePrice.2` should be deleted - we use the MonetaryAmount, not decimals.
The rest should consolidate into one object property hasStikePrice with the parent class `fibo-der-etd-eto:OptionInstrumentExerciseTerms` as domain.

**Comment by ElisaKendall** [comment, marked as answer]
@jziemer2 Hi Jurgen - these are all in provisional ontologies that have not yet been addressed by the DER FCT.  We are just starting to work on cleaning up options over the next couple of months, so hopefully will get to at least some of these issues soon (at least those related to options).

**Comment by JurgenZiemer** [comment]
Thank you @ElisaKendall 
Just a follow-up question, I noticed that FIBO DEV has been shrinking. There were 3094 classes in 2019/Q4 vs. 2765 now in 2020/Q4. Were these classes also from provisional ontologies and removed?

**Comment by ElisaKendall** [comment]
@jziemer2 Yes - mainly due to consolidation of duplicate concepts, concepts that should have been individuals, and rework in derivatives in particular.  We also eliminated all of the classes related to corporate actions messages, as FIBO isn't a process oriented ontology, and to do justice to all of the possible messages related to corporate actions is really out of scope.   We refined and added definitions to all of the securities-specific corporate actions, and dropped the messages.  Those will be added to production sometime over the coming months.  We also spent some time cleaning up the loans hierarchy last year.  It's not complete, but two different groups had tackled loans, and there was a lot of duplication that's now been removed.  You'll see more of that with respect to options over the coming months - there are duplicate classes for OTC vs. exchange traded content, for example.  We've started to tackle some of that recently, but there is more to be done there.

**Comment by ElisaKendall** [comment]
@jziemer2 FYI, a revision we are working on now in the SEC FCT will add back a gazillion classes (I think well over 200, but stay tuned on that), all in the ISO 10962 reference data ontology.  They represent the exhaustive set of classes corresponding to 5 categories under Equities in the CFI standard, with complete coverage for common, preferred, common convertible, preferred convertible, and limited partnership units (for registered shares only, ignoring variants for bearer and 'miscellaneous' forms).  The challenge in following the ups and downs with numbers of classes is that it doesn't account for the "push me pull you" sort of add and subtract processes going on.  I'm not sure what a better metric would be, though, at least not on an annual basis like the comparison you did.  And I haven't figured out the number we are collapsing at the same time, in loans for example, in derivatives (options / forwards), and in asset-backed securities later this year.

**Comment by JurgenZiemer** [comment]
Thanks, @ElisaKendall,
For sharing the FIBO team plans. I don't know if you have statistics on FIBO downloads, but the breakdown of the 1,171 FIB-DM downloads should be similar to your users: 71% of institutional downloads are from banks - only 15% are Investment, Brokerage & Capital Markets.
Hence, the ask I am hearing is for Loans in Production rather than more securities content. The last time a new module entered Production was Derivatives in 2018.
Best regards,
Jurgen
 
![image](https://user-images.githubusercontent.com/11173121/105780673-92ac3780-5fa3-11eb-8673-a58a363ae250.png)

**Comment by rivettp** [comment]
@jziemer2 back to your original complaint, the local names are NOT strictly speaking the same since the properties have initial lower case. That's enough for OWL/RDF to distinguish them: all semantic web and related technologies are case sensitive. If you're aiming to generate data models from other ontologies you won't be able to rely on the "properties start with a verb" convention so you might want to adjust your algorithm. And it's possible even class names might need case to distinguish them.

As a separate point, @ElisaKendall  I'm not a fan of burying something as fundamental to an instrument as strike price in a separate xExerciseTerms resource: why not a direct property of the instrument? xExerciseTerms is not something that a business user would find meaningful: looking at any table of options (for example) the strike price is one of the few properties directly listed along with date and whether it's a Call or Put.

**Comment by jfgemski** [comment]
That’s the problem with statistics, they can be misleading.  Except for small, local banks, most banks are heavily into investment management although they wouldn’t categorize themselves as investment firms since banking is their main charter.  Many of the larger banks even have their own trading desks.

 

For example, did you classify Deutsche Bank (DB) into the Bank category?  That would be a mistake since Wiki has this information about DB:

 

Deutsche Bank is a German multinational  <https://en.wikipedia.org/wiki/Investment_banking> investment bank and  <https://en.wikipedia.org/wiki/Financial_services> financial services company  The company is a  <https://en.wikipedia.org/wiki/Universal_bank> universal bank with three major divisions: the Private & Commercial Bank, the Corporate & Investment Bank (CIB), and Asset Management (DWS).

 

The net effect of the above is that you can’t say that those in the category of Banks aren’t that interested in the FIBO domains already developed or being actively developed.  Yes, a small, local bank probably has much of its business in loans and mortgages but anything larger gets into investment and asset management which involves the areas we’ve already developed in FIBO or are in the process of developing.

John

 

Jfgemski@gmail.com <mailto:Jfgemski@gmail.com>  | M: (516) 353-0004

**Comment by JurgenZiemer** [comment]
Both John (@jfgemski)
2 downloads from https://www.dws.com/ classified as Investment, Brokerage, and FM
5 downloads from https://www.deutsche-bank.de and https://www.db.com as Bank

All online brokers have banking operations, yet we don't classify them as banks. If in doubt, I go by the FI's NAICS (or other standard industry classification).  
https://www.naics.com/company-profile-page/?co=5072

**Comment by jfgemski** [comment]
As I said, they’re chartered as banks and probably originally had most of their business in retail services.  However, there’s only so much performance you can get from retail services.  To attract high asset clients they needed to start investing in securities.  That led to the banks also investing for their own benefit. Thus most large banks, who are probably the bulk of EDMC’s bank members, have significant securities processing operations.

As for DSW, it was recently spun off from DB as a separate unit but still is under the control of DB.  Just like Citibank is now held by Citigroup and Google is now a subsidiary of Alphabet.

So I still say your implied assumption that “bank” just means retail banking with no securities involvement is wrong.

Sent from John's iPad

P.S. Ask John Bottega about this.  If he agrees that “bank” is just retail banking without any securities involvement, I’ll drop this.

**Comment by ElisaKendall** [comment]
@jziemer2 Juergen - our experience on the development side is similar to what John says, above.  Citi joins the SEC calls when they can, and they are really interested in using FIBO as a reference model for securities, among other things.  Loans is on their list as well, but we've been attempting to finalize equities, and address the open issues we know of, so that we can pause on that and move on to something else, possibly asset-backed securities including asset-backed mortgages, which is another area that we have been asked to work on.  For our government partners such as the CFTC, futures and swaps are our highest priority, so we are working on forwards, options, and futures to meet their needs on the derivatives team.  Our use cases and requirements really come from the participants on my calls.  In IND we've been working on cleaning up our representation of any statistical data, which again, has a broad base, and we are close to finanlizing a first pass on security-specific corporate actions, which is in the CAE domain area.  We also recently released our first Funds ontology, which moved the CIV domain into SEC.  So I think the progress is significant and goes beyond what you might think with regards to other domain areas.  We made a choice last yeat to focus on use cases, which is what we've been doing ever since -- if there is an EDMC member with a loans use case, that might motivate work on that area, but no one has suggested / posted a use case for Loans in the wiki to date, or launched a new FCT.  I don't have the bandwidth at the moment unfortunately, unless I have a client supporting the work.
